crates/wdk-build/src/lib.rs

```diff
@@ -1604,12 +1686,246 @@ mod tests {
+
+            // Verify driver-specific logic branches
+            match &driver_config {
+                DriverConfig::Wdm => assert_wdm_usb_headers(&header_contents),
+                DriverConfig::Kmdf(_) => assert_kmdf_usb_headers(&header_contents),
+                DriverConfig::Umdf(_) => assert_umdf_usb_headers(&header_contents),
+            }
+        }
+
+        #[test]
+        fn usb_success() {
+            // Test cases covering usb_headers() logic branches:
+            // 1. WDM - Common headers + kernel-mode headers, no WDF/UCX
+            // 2. KMDF - All headers including WDF, UCX, and KMDF-specific USB
+            // 3. UMDF - Common headers + WDF, no kernel-mode or UCX
+            let test_cases = [
+                (DriverConfig::Wdm, CpuArchitecture::Amd64),
+                (
+                    DriverConfig::Kmdf(KmdfConfig {
+                        kmdf_version_major: 1,
+                        target_kmdf_version_minor: 33,
+                        minimum_kmdf_version_minor: None,
+                    }),
+                    CpuArchitecture::Arm64,
+                ),
+                (
+                    DriverConfig::Umdf(UmdfConfig {
+                        umdf_version_major: 2,
+                        target_umdf_version_minor: 33,
+                        minimum_umdf_version_minor: None,
+                    }),
+                    CpuArchitecture::Amd64,
+                ),
+            ];
+
+            for (driver_config, cpu_architecture) in test_cases {
+                test_usb_headers_with(&driver_config, cpu_architecture);
+            }
+        }
+
+        #[test]
+        fn usb_failure() {
+            // Test KMDF failure case: KMDF drivers require UCX headers, so they fail
+            // when the WDK directory structure is missing or invalid
+            let driver_config = DriverConfig::Kmdf(KmdfConfig {
+                kmdf_version_major: 1,
+                target_kmdf_version_minor: 33,
+                minimum_kmdf_version_minor: None,
+            });
+            let cpu_architecture = CpuArchitecture::Amd64;
+
+            // Create config with nonexistent WDK path to trigger UCX header resolution
+            // failure
+            let config = Config {
+                wdk_content_root: PathBuf::from("/nonexistent/wdk/path"),
+                cpu_architecture,
+                driver_config,
+            };
+
+            let result = config.bindgen_header_contents([ApiSubset::Usb]);
+
+            // KMDF should fail due to UCX header path resolution
+            assert!(
+                result.is_err(),
+                "Expected KMDF USB test to fail with nonexistent WDK path"
+            );
+
+            #[cfg(nightly_toolchain)]
+            assert_matches!(
+                result.unwrap_err(),
+                ConfigError::IoError(_),
+                "Expected IoError due to nonexistent WDK path during UCX header resolution"
+            );
+        }
```

Comment:
I haven't gone through these new tests as I am not fully familiar with the domain logic. Leaving it to @wmmc88 to check them.